api/s2n.h

```diff
@@ -222,6 +222,7 @@ extern uint64_t s2n_connection_get_delay(struct s2n_connection *conn);
 extern int s2n_connection_set_cipher_preferences(struct s2n_connection *conn, const char *version);
 extern int s2n_connection_set_protocol_preferences(struct s2n_connection *conn, const char * const *protocols, int protocol_count);
 extern int s2n_set_server_name(struct s2n_connection *conn, const char *server_name);
+extern int s2n_connection_set_keyshare_by_name(struct s2n_connection *conn, const char* curve_name);
```

Comment:
I _really_ don't like string identifiers but I know it's also consistent with the other functions in the API. Ideally we have an enum instead. What does everyone think here?

My preference would be an enum

Reply:
Though not ideal, I just want to point out that other providers like openssl also use string identifiers for curve names/groups.